include/nbl/builtin/hlsl/scene/keyframe.hlsl

```diff
@@ -0,0 +1,49 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_HLSL_SCENE_KEYFRAME_INCLUDED_
+#define _NBL_HLSL_SCENE_KEYFRAME_INCLUDED_
+
+
+
+#include <nbl/builtin/hlsl/math/quaternions.hlsl>
+
+
+
+namespace nbl
+{
+	namespace hlsl
+	{
+		namespace scene
+		{
+			struct Keyframe_t
+			{
+				uint2 data[3];
+
+				float3 getScale(in Keyframe_t keyframe);
+				quaternion_t getRotation(in Keyframe_t keyframe);
+				float3 getTranslation(in Keyframe_t keyframe);

```

Comment:
great, but note that these methods no longer need the first parameter

Reply:
> great, but note that these methods no longer need the first parameter

The "in" keyword?

Reply:
Oh! Got it!

Reply:
yes, exactly.